src/backends/custom/custom_backend.cc

```diff
@@ -547,6 +545,28 @@ CustomBackend::Context::Run(
     return;
   }
 
+#ifdef TRTIS_ENABLE_STATS
```

Comment:
It seems like all of this could be in InferenceRequest::ReportStats.

Reply:
The batch stats only need to update once per execution, moving into inference request will have duplication or adding a flag to inform which request will trigger the update.

Reply:
Good point, so they will need to remain separate. But I think you can pass in just the timestamps and let the stats aggregator do the math.